### PR TITLE
Generate Enum-Dictionary-Keys (analogous to Newtonsoft)

### PR DESCRIPTION
**Pull request by angelaki**
Fixes https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2820.

Guess that's totally fine since STJ does support enum keys? It probably did not for some time - but I've checked with .Net 6-8

### CODE REVIEW

**Review by martincostello — COMMENTED** [review]
Could you add at least one test for this scenario please?

**Comment by angelaki** [comment]
Just double-checked: dict-key enums are always treated as strings, no converter needed. So I undid the change. Never the less: fix + test here.

**Comment by martincostello** [comment]
Two tests are failing.

**Comment by angelaki** [comment]
Whoops ... that actually was my test, still assuming that with no `StringEnumConverter` the key values stay integers. But since they don't, the assumption was just wrong. Fixed the test.

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:783]: I just took a look, and we don't have an explicit test in this file for a string enum dictionary. Could you add one that's effectively the same as this test for for a dictionary wanting a string enum as the key?

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:781]: This doesn't seem to have an assertion that's integer specific - wouldn't this assertion pass regardless of whether the keys were integers or strings as the number of properties would be the same in both cases?

Comment (angelaki) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:781]: e.g.

{ enumValue1: string, enumValue2: string ...} vs
{ [key: string]: string }

Comment (martincostello) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:781]: In that case, can we assert that the properties are all of type string and have the correct names for that enum?

Comment (angelaki) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:781]: Sure, give me a sec so I compare the names. Maybe not the worst idea so I can check if `options.JsonSerializerOptions.DictionaryKeyPolicy` is respected.

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:782]:
```suggestion
                    configureSerializer: c => c.DictionaryKeyPolicy = camelCaseText ? JsonNamingPolicy.CamelCase : null
```

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs:778]: `expectedPropertyNames`

**Comment by angelaki** [comment]
Since naming strategy isn't a thing related to this feature, I completely removed it from the test now and just compare property names to enum names

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Thanks!

**Comment by angelaki** [comment]
Sure! Thank you for supporting! :)

How long does a deploy of my two PRs usually take?

**Comment by martincostello** [comment]
They will be part of the next release when that's ready. As you can see, there's [still lots of pending PRs](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/milestone/19) and I can't review my own work, so I'm waiting on reviews from others with write access.

**Pull request merged**
